### PR TITLE
feat: Added native props to Select and SelectOption

### LINKED ISSUE #108: Select: add onBlur support and maintain focus
The Select component in iTwinUI-react does not support an onBlur prop & callback. When the drop-down is opened, the Select component seems to lose focus; the focus indicator is no longer around the button. When an option is chosen, focus does not return to the button.

Can the Select component maintain focus while the drop-down is open?
Can support for an onBlur prop & callback be added?

### LINKED ISSUE #225: Add Id props to SelectOption<T>
#### Feature
When writing e2e test, you want to have direct access to element you want to do some actions on it. That mean having some way to click on a specifics item in the select dropdown. I suggest adding Id as optional props in SelectOption<T>.

### LINKED ISSUE #238: Select/MenuItem add value props as data-value={value}
## Feature
Making a big summaries of what is MenuItem, it's an element in a list of elements shown in a dropdown menu. Each MenuItems represent an action or a value that will be send to a function e.g. onClick. From the definition, value is unknown. In reality, value will be a primitive and I wouldn't expect it to be anything else tbh. Value is kept as memory data and is not represent in the html and that is where I see a problem. Select use MenuItem as alternative to option tag. Fine by that as the select tag does not offer much of CSS customization. But here what is missing about the select that old html select offer. **An value attribute.**

## How to satisfy this missing feature ?
Using data-* attributes ! In my case, MenuItem does have the value but is not shown in the html. I'm looking for a way to get the correct item in the select dropdown to make a click action and change the current value. I can use the selector `.iui-select .iui-menu > .iui-menu-item[data-value="SearchValue"]` to find the exact option I want and make a click on the item. All it's need is 2 line added which I can provide in a PR made for Select changes. For the sake of saving my time and yours as this feature request and #225 are alternative to solve the same issue. I will push only one PR.

### PR DESCRIPTION
**Pull request by veekeys**
- Added native div props to Select (checked that Select native props would not work on our component)
- Added `CommonProps` to option and added support to any other prop passed in there

Closes #108, closes #238, closes #225

### CODE REVIEW

File: src/core/Select/Select.tsx [hunk 1]
```diff
@@ -109,7 +113,18 @@ export type SelectProps<T> = {
    */
   popoverProps?: Omit<PopoverProps, 'onShow' | 'onHide' | 'disabled'>;
 } & Pick<PopoverProps, 'onShow' | 'onHide'> &
-  CommonProps;
+  CommonProps &
+  Omit<
+    React.HTMLAttributes<HTMLDivElement>,
```
Comment (mayank99) [inline src/core/Select/Select.tsx:118]: Better to use `React.ComponentPropsWithoutRef<'div'>`

File: src/core/Select/Select.tsx [hunk 2]
```diff
@@ -51,7 +51,11 @@ export type SelectOption<T> = {
    * Item is disabled.
    */
   disabled?: boolean;
-};
+  /**
+   * Any other props.
+   */
+  [key: string]: unknown;
+} & CommonProps;
```
Comment (mayank99) [inline src/core/Select/Select.tsx:58]: This should also be added to `MenuItemProps` then

File: src/core/Select/Select.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Select/Select.tsx:118]: Or maybe even `React.ComponentPropsWithoutRef<'select'>`

File: src/core/Select/Select.tsx (hunk 2 above)
Comment (veekeys) [reply src/core/Select/Select.tsx:58]: only there or keep both?

File: src/core/Select/Select.tsx (hunk 1 above)
Comment (veekeys) [reply src/core/Select/Select.tsx:118]: Select no, I can change to div..
Whats the diff again?

File: src/core/Select/Select.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Select/Select.tsx:58]: might not work if it's only there

File: src/core/Select/Select.tsx (hunk 1 above)
Comment (mayank99) [reply src/core/Select/Select.tsx:118]: Well, for one, you can be explicit about ref.

Second, it's easier to read/remember I guess

**Review by mayank99 — APPROVED** [review]

**Pull request merged**